aptos-move/framework/aptos-framework/sources/jwks.move

```diff
@@ -0,0 +1,703 @@
+
+        let ret = if (found) {
+            option::some(vector::remove(&mut jwks.entries, index))
+        } else {
+            option::none()
+        };
+
+        ret
+    }
+
+    /// Upsert a `JWK` into a `ProviderJWKs`. If this upsert replaced an existing entry, return it.
+    fun upsert_jwk(set: &mut ProviderJWKs, jwk: JWK): Option<JWK> {
+        let found = false;
+        let index = 0;
+        let num_entries = vector::length(&set.jwks);
+        while (index < num_entries) {
+            let cur_entry = vector::borrow(&set.jwks, index);
+            let comparison = compare_u8_vector(get_jwk_id(&jwk), get_jwk_id(cur_entry));
+            if (is_greater_than(&comparison)) {
+                index = index + 1;
+            } else {
+                found = is_equal(&comparison);
+                break
+            }
+        };
+
+        // Now if `found == true`, `index` points to the JWK we want to update/remove; otherwise, `index` points to where we want to insert.
+
+        let ret = if (found) {
+            option::some(vector::remove(&mut set.jwks, index))
+        } else {
+            option::none()
+        };
+
+        vector::insert(&mut set.jwks, index, jwk);
+
+        ret
+    }
+
+    /// Remove the entry of a key ID from a `ProviderJWKs` and return the entry, if exists.
+    fun remove_jwk(jwks: &mut ProviderJWKs, jwk_id: vector<u8>): Option<JWK> {
+        let found = false;
+        let index = 0;
+        let num_entries = vector::length(&jwks.jwks);
+        while (index < num_entries) {
+            let cur_entry = vector::borrow(&jwks.jwks, index);
+            let comparison = compare_u8_vector(jwk_id, get_jwk_id(cur_entry));
+            if (is_greater_than(&comparison)) {
+                index = index + 1;
+            } else {
+                found = is_equal(&comparison);
+                break
+            }
+        };
+
+        // Now if `found == true`, `index` points to the JWK we want to update/remove; otherwise, `index` points to where we want to insert.
+
+        let ret = if (found) {
+            option::some(vector::remove(&mut jwks.jwks, index))
+        } else {
+            option::none()
+        };
+
+        ret
+    }
+
+    /// Modify an `AllProvidersJWKs` object with a `Patch`.
+    fun apply_patch(jwks: &mut AllProvidersJWKs, patch: Patch) {
+        let variant_type_name = *string::bytes(copyable_any::type_name(&patch.variant));
+        if (variant_type_name == b"0x1::jwks::PatchRemoveAll") {
+            jwks.entries = vector[];
+        } else if (variant_type_name == b"0x1::jwks::PatchRemoveIssuer") {
+            let cmd = copyable_any::unpack<PatchRemoveIssuer>(patch.variant);
+            let (found, index) = vector::find(&jwks.entries, |obj| {
+                let provider_jwk_set: &ProviderJWKs = obj;
+                provider_jwk_set.issuer == cmd.issuer
+            });
+            if (found) {
+                vector::swap_remove(&mut jwks.entries, index);
```

Comment:
Ehh, this will break the ordering on `AllProvidersJWKs`, no?

Reply:
thanks for another catch! it's supposed to be simply calling `remove_issuer()`

Reply:
I changed it to `remove`.

Reply:
Oh, even better. Changing to that!